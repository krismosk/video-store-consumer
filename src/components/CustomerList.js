import React, { Component } from 'react';
import Customer from './Customer.js';
import axios from 'axios';
import PropTypes from 'prop-types';

class CustomerList extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      customers: [],
    };
  }

  componentDidMount () {
    axios.get('http://localhost:3000/customers')
      .then((response) => {
        this.setState({
          customers: response.data,
        });
      })
      .catch((error) => {
        console.log(error.message);
      });
  }


  findCustomer = (customerId) => {
    const selectedCustomer = this.state.customers.find((customer) => {
      return customer.id === customerId;
    })
    
    this.props.selectCustomer(selectedCustomer);
  }

  makeCustomerList() {
   const customerList = this.state.customers.map((customer, i) => {
      return <Customer
        key={ i }
        id={ customer.id }
        name={ customer.name }
        address={ customer.address }
        city={ customer.city }
        state={ customer.state }
        postalCode={ customer.postalCode }
        phone={ customer.phone }
        findCustomer={ this.findCustomer }
      />
    })

    return customerList;
  }

  render () {
    return (
      <div>
        { this.makeCustomerList() }
      </div>
    )
  }
}

CustomerList.propTypes = {
  selectCustomer: PropTypes.func.isRequired
}

export default CustomerList;