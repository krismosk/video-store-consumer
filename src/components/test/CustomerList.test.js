import React from 'react';
import { render, cleanup } from '@testing-library/react';
import CustomerList from '../CustomerList';

const customerList = <CustomerList 
  selectCustomer = {() => { }}
/>;

describe("Customer List", () => {
  test('it can be rendered and matches the snapshot', () => {
    const { asFragment } = render ( customerList );

    expect(asFragment()).toMatchSnapshot();
    cleanup();
  });
})