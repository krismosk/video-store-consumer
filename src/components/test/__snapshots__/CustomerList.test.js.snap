// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Customer List it can be rendered and matches the snapshot 1`] = `
<DocumentFragment>
  <div />
</DocumentFragment>
`;
